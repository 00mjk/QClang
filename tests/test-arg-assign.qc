qubit foo(qubit q)
{
	return !q;
}

int main() {
	qubit q;
	
	q = foo(false);
	measure(q);
	return 0;
}

