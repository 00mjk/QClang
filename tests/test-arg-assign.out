OPENQASM 2.0;
include "qelib1.inc";
qreg q[1];
creg c[1];
h q;
creg tmp1_bt0[1];
qreg tmp0_qt0[1];
qreg q_q0[1];
qreg temp_1[1];
x temp_1[0];
creg temp_1_mb2[1];
measure temp_1 -> temp_1_mb2;
